apps-client/mobile/components/poll/poll-card.tsx

```diff
@@ -0,0 +1,353 @@
+            id={option.id}
+            title={option.title}
+            votes={1} // TODO: integrate with real votes
+            totalVotes={2} // TODO: integrate with real total votes
+            isSelected={optionIds.includes(option.id)}
+            pollTouched={optionIds.length > 0}
+            onSelect={handleOptionSelect}
+          />
+        ))}
+    </>
+  )
+}
+
+interface PollHeaderProps {
+  poll: PollItem
+  isEnded: boolean
+  triggerEnded: () => void
+}
+
+const PollHeader = (props: PollHeaderProps) => {
+  return (
+    <View className="gap-1 flex flex-col mb-2">
+      <View className="w-full flex flex-row items-center gap-2">
+        <Icon
+          icon={MessageCircleQuestionIcon}
+          size={32}
+          className="text-base-primary-default"
+          strokeWidth={2}
+        />
+        <Text className="font-body-medium flex-1">{props.poll.title}</Text>
+      </View>
+      <View className="flex flex-row items-center justify-between">
+        <View className="flex flex-row items-center">
+          <Icon
+            icon={Clock3Icon}
+            size={16}
+            className="text-base-text-medium mr-1"
+            strokeWidth={1}
+          />
+          <Text className="text-base-text-medium font-heading-regular text-sm items-center mt-2">
+            เวลาที่เหลือ:{' '}
+            <Text
+              className={cn(
+                'text-base-primary-default font-heading-bold',
+                props.isEnded && 'text-base-text-high'
+              )}
+            >
+              <PollCountdownTimer triggerEnded={props.triggerEnded} targetTime={props.poll.endAt} />
+            </Text>
+          </Text>
+        </View>
+        <PollStatusBadge isEnded={props.isEnded} />
+      </View>
+    </View>
+  )
+}
+
+interface PollCountdownTimerProps {
+  targetTime: Date
+  triggerEnded: () => void
+}
+
+export function PollCountdownTimer(props: PollCountdownTimerProps) {
+  const [display, setDisplay] = React.useState('')
+
+  React.useEffect(() => {
+    const timer = setInterval(() => {
+      const now = dayjs()
+      const target = dayjs(props.targetTime)
+      const diff = target.diff(now)
+
+      if (diff <= 0) {
+        setDisplay('หมดเวลา')
+        clearInterval(timer)
+        props.triggerEnded()
+        return
+      }
+
+      const d = dayjs.duration(diff)
+      const totalHours = d.asHours()
+      const totalMinutes = d.asMinutes()
+      const totalSeconds = d.asSeconds()
+
+      if (totalHours >= 24) {
+        setDisplay(`${Math.floor(d.asDays())} วัน`)
+      } else if (totalHours >= 1) {
+        setDisplay(`${Math.floor(totalHours)} ชั่วโมง`)
+      } else if (totalMinutes >= 1) {
+        setDisplay(`${Math.floor(totalMinutes)} นาที`)
+      } else {
+        setDisplay(`${Math.floor(totalSeconds)} วินาที`)
+      }
+    }, 1000)
+
+    return () => clearInterval(timer)
+  }, [props.targetTime, props.triggerEnded, props])
+
+  return <>{display}</>
+}
```

Comment:
rendering logic should be down below

```suggestion
       setDiff(diff)
    }, 1000)

    return () => clearInterval(timer)
  }, [props.targetTime, props.triggerEnded])

  return <>{formatTimeInterval(diff)}</>
}
// extract format function
function formatTimeInterval(diff: number) {
  const d = dayjs.duration(diff)
  const totalHours = d.asHours()
  const totalMinutes = d.asMinutes()
  const totalSeconds = d.asSeconds()

  if (totalHours >= 24) {
    return `${Math.floor(d.asDays())} วัน`
  } else if (totalHours >= 1) {
    return`${Math.floor(totalHours)} ชั่วโมง`
  } else if (totalMinutes >= 1) {
    return`${Math.floor(totalMinutes)} นาที`
  } else {
    return`${Math.floor(totalSeconds)} วินาที`
  }
}
```